gen/src/Gen/Tree.hs

```diff
@@ -47,7 +43,7 @@ fold ::
 fold g f (p :/ t) = (p :/) <$> go p t
   where
     go x = \case
-      Failed n e -> fail (show e) >> return (Failed n e)
+      Failed n e -> fail (show e) >> pure (Failed n e)
```

Comment:
```suggestion
      Failed n e -> fail (show e)
```
Because `fail (show e) >> pure (Failed n e)` = `fail (show e) >>= \_ -> pure (Failed n e)` = `fail (show e)` by the `MonadFail` laws?

Reply:
Organically grown grass-fed code.